modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -12490,4 +12491,44 @@ public static long uncompressedSize(File zip) throws IOException {
     public static String enabledString(boolean enabled) {
         return enabled ? "enabled" : "disabled";
     }
+
+    /**
+     * Tries to increase pool of empty pages in data region, if it is too small to process large entries.
+     *
+     * @param ctx Cache context.
+     * @param e Exception caused by {@link IgniteOutOfMemoryException}.
+     * @return Exception if the increasing of the pool has failed, {@code null} otherwise.
+     */
+    @Nullable public static IgniteCheckedException preventOutOfMemoryOperationFailure(GridCacheContext ctx, Exception e) {
+        if (!ctx.dataRegion().increaseEmptyPagesPool()) {
+            return new IgniteCheckedException("Failed to increase empty pool size [" +
+                "currentSize=" + ctx.dataRegion().emptyPagesPoolSize() +
+                ", dataRegion=" + ctx.dataRegion().config().getName() +
+                ", emptyPoolSize=" + ctx.dataRegion().emptyPagesPoolSize() +
```

Comment:
I think that this line should add the value specified at DataRegionConfiguration. `ctx.dataRegion().emptyPagesPoolSize()` is already logged as `currentPoolSize`.